src/server/main_service.cc

```diff
@@ -1246,10 +1247,16 @@ void Service::DispatchManyCommands(absl::Span<CmdArgList> args_list,
     dfly_cntx->transaction = dist_trans.get();
     MultiCommandSquasher::Execute(absl::MakeSpan(stored_cmds), dfly_cntx, this, true, false);
     dfly_cntx->transaction = nullptr;
+
+    dispatched += stored_cmds.size();
     stored_cmds.clear();
   };
 
   for (auto args : args_list) {
+    // Stop accumulating when a pause is requested, fall back to regular dispatch
+    if (dfly::ServerState::tlocal()->IsPaused())
+      break;
```

Comment:
The current logic of await dispatches is the following: Every command that is dispatching during paused = false, will eventually conclude and will reply to our checkpoint request

That is why we have to abort early here to handle the checkpoint, and then we invoke non-squashed dispatch for the next command to become paused

Reply:
Do we have to break here or we can change the logic to pause inside InvokeCmd instead inside DispatchCommand and this way we will pause on squashed commands as well, the same change that I did in my last PR

Reply:
I prefer breaking here because I don't know how to easily solve it the other way round.

As long as DispatchManyCommands is running, we have `async_dispatch=true`, even when it just switches commands. This implementation doesn't block clients, so we can only track running commands when we set the pause state. If we set paused during pipeline squashing and let it block, we will wait for it forever

We don't have this problem now, because we:
* block clients
* wait for squashing to finish (with `AwaitCurrentDispatches`)
* don't have 100% guarantees with `AwaitCurrentDispatches`. We currently have two balancing `ThisFiber::Yield` calls in `dragonfly_connection`, so  `AwaitCurrentDispatches` might see the connection fiber sleeping and not dispatching, even if it really has more work to do... And by the time we reply OK to the client, we could still be running commands that we missed (+ to https://github.com/dragonflydb/dragonfly/pull/2179#discussion_r1395600349)